File: 2022/swift/01.swift
import Foundation

func main(_ filename: String) {
    guard let contents = try? String(contentsOfFile: filename) else {
        print("Failed to open \(filename)")
        return
    }

    let blocks = contents.components(separatedBy: "\n\n").map { $0.split(separator: "\n").map { Int($0)! }}

    let cals = blocks.reduce([Int]()) {
        prev, curr in
        var next = prev
        next.append(curr.reduce(0, +))

        return next
    }.sorted().reversed().map { $0 }

    print("Part one:", cals[0])
    print("Part two:", cals[0 ... 2].reduce(0, +))
}

let filename = "./input/01.in"

main(filename)

File: 2022/deno/01.ts
const input = await Deno.readTextFile("./input/01.in");

const data = input.split("\n\n").map((chunk) => chunk.split("\n").map(Number));

/**
 * Part One
 */
const sum = (a: number, b: number) => a + b;

const cals = data
  .reduce((prev, curr) => {
    const total = curr.reduce(sum);
    return [...prev, total];
  }, [])
  .sort((a, b) => b - a);

console.log("Part one:", cals[0]);

/**
 * Part Two
 */
console.log("Part two:", cals.slice(0, 3).reduce(sum));
